### PR TITLE
feat(taiko-client): add optional `l1.private`

### PR DESCRIPTION
**Pull request by YoGhurt111**

### CODE REVIEW

File: packages/taiko-client/cmd/flags/common.go [hunk 1]
```diff
@@ -28,6 +28,12 @@ var (
 		Category: commonCategory,
 		EnvVars:  []string{"L1_WS"},
 	}
+	L1PrivateEndpoint = &cli.StringFlag{
+		Name:     "l1.private",
```
Comment (dantaik) [inline packages/taiko-client/cmd/flags/common.go:32]: private_tx_endpoint

**Review by mask-pp — APPROVED** [review]

File: .github/workflows/taiko-client--docker.yml [hunk 2]
```diff
@@ -2,7 +2,7 @@ name: "Push docker image to GAR"
 
 on:
   push:
-    branches: [main]
+    branches: [feature/private-tx-endpoint]
```
Comment (davidtaikocha) [inline .github/workflows/taiko-client--docker.yml:5]: Lets change it back?

File: packages/taiko-client/cmd/flags/common.go [hunk 3]
```diff
@@ -28,6 +28,12 @@ var (
 		Category: commonCategory,
 		EnvVars:  []string{"L1_WS"},
 	}
+	L1PrivateTxEndpoint = &cli.StringFlag{
+		Name:     "l1.privateTx",
```
Comment (davidtaikocha) [inline packages/taiko-client/cmd/flags/common.go:32]:
```suggestion
		Name:     "l1.privateTxWs",
```

File: packages/taiko-client/cmd/flags/common.go [hunk 4]
```diff
@@ -28,6 +28,12 @@ var (
...
+		Usage:    "Websocket RPC endpoint of a L1 private ethereum node",
+		Category: commonCategory,
+		EnvVars:  []string{"L1_PRIVATE_TX"},
```
Comment (davidtaikocha) [inline packages/taiko-client/cmd/flags/common.go:35]:
```suggestion
		EnvVars:  []string{"L1_PRIVATE_TX_WS"},
```

File: packages/taiko-client/proposer/config.go [hunk 5]
```diff
@@ -105,5 +106,10 @@ func NewConfigFromCliContext(c *cli.Context) (*Config, error) {
 			l1ProposerPrivKey,
 			c,
 		),
+		PrivateTxmgrConfigs: pkgFlags.InitTxmgrConfigsFromCli(
+			c.String(flags.L1PrivateTxEndpoint.Name),
```
Comment (davidtaikocha) [inline packages/taiko-client/proposer/config.go:110]: This value might be an empty string, is that ok?

File: .github/workflows/taiko-client--docker.yml (hunk 2 above)
Comment (YoGhurt111) [reply .github/workflows/taiko-client--docker.yml:5]: Will change it back when all test are finished in hekla.

File: packages/taiko-client/cmd/flags/common.go (hunk 3 above)
Comment (YoGhurt111) [reply packages/taiko-client/cmd/flags/common.go:32]: Both http and ws are supposed to be supported.

File: packages/taiko-client/cmd/flags/common.go (hunk 4 above)
Comment (YoGhurt111) [reply packages/taiko-client/cmd/flags/common.go:35]: Both http and ws are supposed to be supported.

File: packages/taiko-client/cmd/flags/common.go (hunk 3 above)
Comment (davidtaikocha) [reply packages/taiko-client/cmd/flags/common.go:32]: But in flag description it says websocket endpoint?

File: packages/taiko-client/proposer/config.go (hunk 5 above)
Comment (YoGhurt111) [reply packages/taiko-client/proposer/config.go:110]: It is ok . 
Added `if cfg.PrivateTxmgrConfigs != nil && len(cfg.PrivateTxmgrConfigs.L1RPCURL) > 0` when it was used

File: packages/taiko-client/cmd/flags/common.go (hunk 3 above)
Comment (YoGhurt111) [reply packages/taiko-client/cmd/flags/common.go:32]: Let me fix the description!

**Review by davidtaikocha — APPROVED** [review]

**Pull request merged**